File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int count = 0;
    public int solution(int[] numbers, int target) {
        int answer = 0;
        dfs(numbers, 0, target, 0);
        answer = count;
        
        return answer;
    }
    
    public void dfs(int[] numbers, int depth, int target, int sum) {
        if (depth == numbers.length) {
            if (target == sum) {
                count++;
            }
            return;
        }
        
        int plus = sum + numbers[depth];
        int minus = sum - numbers[depth];
        
        dfs(numbers, depth+1, target, plus);
        dfs(numbers, depth+1, target, minus);
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(numbers, target, depth, total):
    global count
    
    if(depth == len(numbers)):
        if(target == total):
            count += 1
        return 
    
    plus = total + numbers[depth]
    minus = total - numbers[depth]
    
    dfs(numbers, target, depth+1, plus)
    dfs(numbers, target, depth+1, minus)
    
def solution(numbers, target):
    global count
    count = 0
    dfs(numbers, target, 0, 0)
    return count